packages/codecs-core/README.md

```diff
@@ -362,6 +362,31 @@ const get32BytesBase58Decoder = () => fixDecoder(getBase58Decoder(), 32);
 const get32BytesBase58Codec = () => combineCodec(get32BytesBase58Encoder(), get32BytesBase58Codec());
 ```
 
+## Prefixing the size of codecs
+
+The `prefixCodec` function allows you to store the byte size of any codec as a number prefix. This allows you to contain variable-size codecs to their actual size.
```

Comment:
I wonder if there's a name we can give this codec to make it more clear that the prefix it produces is _particularly_ the size of the field.

I wonder what other libraries call this.

* Length-delimited records ([protobuf](https://protobuf.dev/programming-guides/encoding/#length-types))
* Length-prefix framing

Doesn't super matter. I can't find strong evidence that there's even a consensus on what to call such a thing.

Reply:
Yeah naming this one has been tricky. The constraints for naming this were:

- Needs to start with a verb and finish with `Codec` since it's a codec primitive in the likes of `mapCodec` or `fixCodec`.
- Somewhat echo the `fixCodec` primitive which makes any codec a fixed-size one by providing the byte size explicitly. This is because technically, both of these codecs are responsible for "sizing" the codec they wrap.

`prefixCodec` seemed to do the job okay but I agree it's not ideal. My main issues with it are:

- Not clear that it's prefixing the size.
- A bit too close to the new `getHiddenPrefixCodec` helper at the end of this stack.

An alternative could be `prefixCodecSize` and we could rename `fixCodec` to `fixCodecSize` to bring the two closer together.

Wdyt?

Reply:
I made the suggested change in a subsequent PR: https://github.com/solana-labs/solana-web3.js/pull/2411.